tests/logger.test.ts

```diff
@@ -0,0 +1,81 @@
+import chalk from "chalk";
+import logger from "../src/shared/logger";
+
+const mocks = {
+  error: jest.spyOn(console, "error").mockImplementationOnce(() => {}),
+  log: jest.spyOn(console, "log").mockImplementationOnce(() => {}),
+  info: jest.spyOn(console, "info").mockImplementationOnce(() => {}),
+  warn: jest.spyOn(console, "warn").mockImplementationOnce(() => {}),
+  exit: jest
+    .spyOn(process, "exit")
+    // @ts-ignore - eslint won't allow assertion of `code as never`
+    .mockImplementationOnce(code => code),
+};
+
+afterEach(() => {
+  jest.resetAllMocks();
+  jest.resetModules();
+});
+
+describe('NODE_ENV === "test"', () => {
+  describe.each(Object.keys(logger))("logger.%s", name => {
+    test(`stops execution of logger.${name} if in a test environment`, () => {
+      logger[name]("a test message");
+      expect(mocks.error).toBeCalledTimes(0);
+      expect(mocks.info).toBeCalledTimes(0);
+      expect(mocks.log).toBeCalledTimes(0);
+      expect(mocks.warn).toBeCalledTimes(0);
+      expect(mocks.exit).toBeCalledTimes(0);
+    });
+  });
+});
+
+describe('NODE_ENV === "production"', () => {
+  beforeEach(() => {
+    process.env.NODE_ENV = "production";
+  });
+
+  const message = "a test message!";
+  const table = [
+    ["info", chalk.green.bold("INFO: ") + message],
+    ["warn", chalk.yellow.bold("WARN: ") + message],
+    ["log", message],
+  ];
+
+  describe.each(table)("logger.%s", (name, msg) => {
+    test(`calls "console.${name}()" with "${msg}"`, () => {
+      logger[name](message);
+      const mock = mocks[name];
+      expect(mock).toBeCalledTimes(1);
+      expect(mock).toBeCalledWith(msg);
+    });
+  });
+
+  describe(`logger.error`, () => {
+    it('calls "console.error()" when passed an error instances', () => {
+      const error = new Error();
```

Comment:
```suggestion
      const error = new Error("test error");
```
So we are sure it logs the given error and not just a random one.